### ISSUE TITLE
--generate-cmd secretsmanager YAML not well-formed

### ISSUE THREAD

**Issue by hirokihokari** [aws/copilot-cli#3458]
### Reproduction

1. clone https://github.com/hirokihokari/copilot-test
2. `copilot app init` # as "copilot-nginx"
3. `copilot init`
4. `copilot deploy`
5. `copilot task run --generate-cmd copilot-nginx/test/nginx`
6. run the generated command
```
provision resources for task copilot-test-nginx: check if changeset is empty: create change set copilot-<some id> for stack task-copilot-test-nginx: ValidationError: Template format error: YAML not well-formed. (line 72, column 120)
	status code: 400, request id: <some id>: describe change set copilot-<some id> for stack task-copilot-test-nginx: ChangeSetNotFound: ChangeSet [copilot-<some id>] does not exist
	status code: 404, request id: <some id>
```

7. Remove the secrets block from `manifest.yml`
8. `copilot deploy`
9. `copilot task run --generate-cmd copilot-nginx/test/nginx`
10. succeeds

### double colon?
If I remove the double colon at the end of secret arn, this error goes away and it switches to 
```
ResourceInitializationError: ... secrets manager: failed to retrieve secret from arn:aws:secretsmanag...
```

Is that the right format for secret manager arn?

I can't tell either syntax is wrong (trying to look for mal-formed arn=non-existent secret), or #3256 is not in effect, since the error message is truncated in the terminal as well as [on console](https://github.com/aws/containers-roadmap/issues/1133)

**Comment by huanjani** [comment]
Hello, @hirokihokari!
I tried to reproduce your error, but wasn't able to; I looked at your manifest file in your repo and it looks like it's formatted correctly. Is your Secrets Manager secret already in existence and tagged?
Which version of Copilot are you running?
Thanks!

**Comment by hirokihokari** [comment]
@huanjani 

Hi.
Thank you for the attention.
Could be something about my aws environment...

Secrets Manager secret is there and tagged with `copilot-application` and `copilot-environment` each with its proper value. It's still possible that I've misspelled it in the config though.

per copilot version,
```
$ copilot --version
copilot version: v1.16.0
```


Then I've looked at below example in [Secret](https://aws.github.io/copilot-cli/docs/developing/secrets/) page
```
  # Option 2. Alternatively, you can refer to the secret by ARN.
  DB: "'arn:aws:secretsmanager:us-west-2:111122223333:secret:demo/test/mysql-Yi6mvL'"
```

Noticed a double quote and single quote wrapping arn, tried with this style, and it worked.
It worked, as in no error from task, and confirmed the task has access to the secrets (I haven't tried all secrets, but all "types" of secrets)

i.e.
```
copilot task run \
 ...
--secrets SECRET="'arn:aws ... SECRET::'" \
 ...
```

Does this mean `--generate-cmd` is generating incorrectly, or my manifest.yml and my deploy, which I think `--generate-cmd` bases its configuration from?

**Comment by hirokihokari** [comment]
@huanjani
belated but the `--generate-cmd` outputs secrets like this without quotes
```
--secrets SECRET=arn:aws: ...
```

Perhaps this generation is wrong caused by something in my aws environment?

**Comment by huanjani** [comment]
Thank you for finding this bug! We need the double quotes around the ARN in yaml because of the colons...I am digging into the effect on `--generate-cmd` behavior.

We will fix this asap! Thanks again!

**Comment by huanjani** [comment]
@hirokihokari: Thanks again for opening this issue. The fix is now released in v1.18.0: https://github.com/aws/copilot-cli/releases/tag/v1.18.0!
